src/main/java/org/jenkins/tools/test/model/hook/PluginCompatTesterHooks.java

```diff
@@ -200,11 +199,8 @@ private Map<String, Queue<PluginCompatTesterHook>> findHooks(String stage) {
                         allForType.add(hook);
                         sortedHooks.put(plugin, allForType);
                     }
-                } catch (Exception ex) {
-                    LOGGER.log(
-                            Level.WARNING,
-                            "Error when loading " + c.getName() + "; continuing",
-                            ex);
+                } catch (ReflectiveOperationException e) {
+                    throw new RuntimeException("Error when loading " + c.getName(), e);
```

Comment:
This should never happen, and if it does it is a serious error that would invalidate our confidence in the test run, so throw it up and halt execution.